### PR TITLE
feat-wishlist functionality added

### PR DESCRIPTION
**Pull request by aishwarya-bm**
feat-wishlist functionality added
add,removefrom wishlist, move wishlist to cart & cart to wishlist

https://user-images.githubusercontent.com/66695068/160300019-771a4929-944a-411e-8334-726156860314.mp4

### CODE REVIEW

File: src/components/navpills/Navpills.jsx [hunk 1]
```diff
@@ -25,9 +26,9 @@ export default function Navpills() {
               <Link to="/wishlist">
                 <i className="fa fa-solid fa-heart nav-btn"></i>
               </Link>
-              {/* {wishlistState.size > 0 && (
-                <span className="badge">{wishlistState.size}</span>
-              )} */}
+              {stateWishlist.wishlistSize > 0 && (
+                <span className="badge">{stateWishlist.wishlistSize}</span>
+              )}
```
Comment (ttahm3d) [inline src/components/navpills/Navpills.jsx]: The Idea behind using context is to abstract away the render logic away from componenets.  Even though you are not using anything apart from what is already there in context, you can create another function that returns the wishlistsize in context and pass it down the context through values

File: src/contexts/wishlistContext/wishlist-context.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+const WishlistContext = createContext();
+
+const WishlistProvider = ({ children }) => {
+  const [stateWishlist, dispatchWishlist] = useReducer(wishlistReducer, {
+    wishlist: [],
+    wishlistSize: 0,
+  });
+  return (
+    <>
+      <WishlistContext.Provider value={{ stateWishlist, dispatchWishlist }}>
```
Comment (ttahm3d) [inline src/contexts/wishlistContext/wishlist-context.jsx]: what i meant in previous comment was 

```JSX
<WishlistContext.Provider value={{ 
stateWishlist, 
dispatchWishlist,
wishlistSize: stateWishlist.wishlistSize
}}>

```

**Review by ttahm3d — COMMENTED** [review]
Everything looks good.

I had one suggestion which is mentioned in comments.

Please split your commits. It is not a good practice to commit everything at once.

**Review by sadathshariff — APPROVED** [review]
The Code looks Clean and Good

File: src/components/navpills/Navpills.jsx (hunk 1 above)
Comment (aishwarya-bm) [reply src/components/navpills/Navpills.jsx]: Thanks for this, my code looks a lot cleaner

File: src/contexts/wishlistContext/wishlist-context.jsx (hunk 2 above)
Comment (aishwarya-bm) [reply src/contexts/wishlistContext/wishlist-context.jsx]: made the changes for both wishlist & cart

**Review by ttahm3d — APPROVED** [review]

**Pull request merged**